txpipe/lssweights.py

```diff
@@ -0,0 +1,1103 @@
+			Fmap_bf.update_values_pix(self.sys_maps[0].valid_pixels, np.ones(len(self.sys_maps[0].valid_pixels)).astype(np.float64) )
+
+			fit_output = {}
+			fit_output['sig_map_index'] = sig_map_index
+			fit_output['coeff'] = None
+			fit_output['coeff_cov'] = None
+
+		else: #there are maps to correct
+			print('{0} map(s) selected for correction'.format(len(sig_map_index)))
+			sysmap_table_selected = sysmap_table_all[sig_map_index]
+
+			#make the delta_g map and fracdet weights
+			deltag, frac = self.get_deltag(density_correlation.tomobin)
+			dg1 = deltag[self.sys_maps[0].valid_pixels]+1.0 #deltag+1 for valid pixels
+			weight = frac[self.sys_maps[0].valid_pixels] #weight for samples (prop to 1/sigma^2)
+
+			#do linear regression
+			if self.config["regression_class"].lower() == "linearregression":
+				reg = sk_linear_model.LinearRegression()
+				reg.fit(sysmap_table_selected.T, dg1, sample_weight=weight )
+			elif self.config["regression_class"].lower() == "elasticnetcv":
+				reg = sk_linear_model.ElasticNetCV()
+				reg.fit(sysmap_table_selected.T, dg1, sample_weight=weight )
+			else:
+				raise IOError("regression method {0} not yet implemented".format(self.config["regression_class"]))
+
+			#get output of regression
+			Fvals = reg.predict(sysmap_table_selected.T)
+			Fmap_bf = hsp.HealSparseMap.make_empty(self.sys_maps[0].nside_coverage, self.sys_maps[0].nside_sparse, dtype=np.float64, sentinel=hp.UNSEEN)
+			Fmap_bf.update_values_pix(self.sys_maps[0].valid_pixels, Fvals.astype(np.float64) )
+			coeff = np.append(reg.intercept_,reg.coef_)
+			coeff_cov = None
+
+			#make the 1d trends for the regression output
+			Fdc_bf = lsstools.DensityCorrelation()
+			Fdc_bf.set_precompute(density_correlation)
+			for imap, sys_vals in enumerate(sysmap_table_all):
+				edges = density_correlation.get_edges(imap)
+				Fdc_bf.add_correlation(imap, edges, sys_vals, Fmap_bf[Fmap_bf.valid_pixels], frac=frac[Fmap_bf.valid_pixels], map_input=True, use_precompute=True )
+			density_correlation.add_model(Fdc_bf.ndens, 'linear')
+
+			#assemble the fitting outputs (chi2, values, coefficents, )
+			fit_output = {}
+			fit_output['sig_map_index'] = sig_map_index
+			fit_output['coeff'] = coeff
+			fit_output['coeff_cov'] = coeff_cov
+
+		f = time.time()
+		print("compute_weights took {0}s".format(f-s))
+
+		return Fmap_bf, fit_output
+
+
+
+def hsplist2array(hsp_list):
+	"""
+	Convert a list of healsparse maps to a 2d array of the valid pixels
+	Assume all maps have the same mask
+	"""
+	import numpy as np 
+	validpixels = hsp_list[0].valid_pixels
+	out_array = []
+	for i in range(len(hsp_list)):
+		out_array.append(hsp_list[i][validpixels])
+	return np.array(out_array)
+
+
+
+class TXLSSweightsUnit(TXLSSweights):
+	"""
+	Class assigns weight=1 to all lens objects
+
+	"""
+	name = "TXLSSweightsUnit"
+	parallel = False
+
```

Comment:
I think you can override the inputs and outputs here since many of them are not made.  You might not even want this to descend from `TXLSSWeights` if it doesn't share much behaviour?

Reply:
The `save_weights` method from `TXLSSWeights` still gets used which uses the same inputs and output, so I think we can keep this as is?